src/generators.jl

```diff
@@ -883,27 +882,96 @@ function forest_fire_graph(::Type{S},seed_network::Union{MatrixNetwork,SparseMat
         parent = random_node(graph_size)
         
         new_v = graph_size+1
-        _burn!(neighbors,new_v,parent,burn_p,
-                     toburn,burnt,burning,alive)
+        burn!(neighbors,new_v,parent,burn_p,
+              toburn,burnt,burning,alive;rng)
 
         parents[v_i] = parent
-        for v_j in neighbors[new_v]
-            push!(neighbors[v_j],new_v)
-        end 
         graph_size += 1
 
     end 
     
-    A =  list_of_list_to_sparse_matrix(neighbors)
-    if typeof(seed_network) === MatrixNetwork
-        A = MatrixNetwork(A)
-    end
-    return A, parents
+    return list_of_list_to_matrix(typeof(seed_network),neighbors), parents
+end 
+
+"""
+`burn`
+======
+
+produce an edge list of the vertices randomly traversed (a.k.a "burning") 
+starting from `v0` in a graph `A`. The number of edges selected at each 
+step is determined by a geometric distribution with success rate `p`.
+
+Input
+-----
+- 'A::Union{MatrixNetwork,SparseMatrixCSC}': An undirected seed Network. 
+- 'v0::Int': The starting node of the random walk. 
+- 'p::Float64': The geometric distribution's success probability for selecting
+  the edges in the random walk. 
+
+Output
+------
+- 'neighbors::Vector{S}': The final produced neighbor list from the burn 
+  process. 
+
+See also [`forest_fire_graph`](@ref), [`burn!`](@ref)
+""" 
+function burn(A::Union{MatrixNetwork,SparseMatrixCSC},v0::S,p;kwargs...) where S
+
+    neighbors = matrix_to_list_of_list(A)
+    push!(neighbors,Vector{S}(undef,0))
+    burn!(neighbors,A.n+1,v0,p;kwargs...)
+    return neighbors
+
 end 
 
 
-function _burn!(neighbors::Vector{Vector{S}},new_v::Int,v0::Int,burn_p::Float64,
-                      toburn,burnt,burning,alive) where S 
+"""
```

Comment:
Sorry if there was any misunderstanding, burn should not _add_ the edges to the graph. It should simply return the the set of burned vertices. e.g. if you wanted to run a forest fire sample of an existing graph, you want to run "burn" from multiple vertices. This is a sampled graph that should have similar properties to the original. But you definitely don't want to add anything.

Reply:
Taken care of! Any suggestions on tests that can be added to this routine? Currently just ensuring seeded calls return the same for MatrixNetwork and SparseMatrixCSC inputs.